### ISSUE TITLE
Add GitHub Action for build after merge

### ISSUE THREAD

**Issue by loffa** [itsektionen/songlist#16]
I suggest we add a build job in the GitHub actions to build after merges to master branch to make sure that dist is always up to date with changes.

**Comment by Cactooz** [comment]
This sounds like a great idea!

**Comment by StelFoog** [comment]
This is a great idea! 😄 I tried doing this before (generating a release for each push) [here](https://github.com/StelFoog/songlist-test). I had trouble programmatically getting files from the latest release, seemingly due to GitHub's cors policy. If there are any ideas on how to resolve it (or get around it) then I think it's the most obvious improvement that could be made!

**Comment by Cactooz** [comment]
Do we want it to:
1. Build and release if something gets merged into main
2. PR gets built and saved into that PR branch that then can be merged into main, without making a release

**Comment by Cactooz** [comment]
> An alternative would be to build in main and make the action do a commit with the files, like what is done manually now.

Do you mean that we push unbuilt and then the action goes and builds and pushes a new commit directly to main. Ending up with 2 commits in total?

**Comment by loffa** [comment]
Yep, if that makes it easier to just download raw files from the dist repo

**Comment by Cactooz** [comment]
Yeah it could work. As long as you don't download it while the build is happening. Then it would be broken.

The one that does the action in the PR will make sure that the merged commit is built and ready directly.

**Comment by StelFoog** [comment]
> An alternative would be to build in main and make the action do a commit with the files, like what is done manually now.

This is good workaround! The best option would still be building and keeping the files separate from the codebase, but until we have such a solution this is far better than having to manually do it.

Could it be possible to keep the builds on a separate branch?

**Comment by Cactooz** [comment]
> Could it be possible to keep the builds on a separate branch?

It should be possible with some trickery using the GitHub CLI in the GitHub Actions, where you can checkout and push to other branches. The only issue is that someone has to generate a GitHub token and put that in the repo for the Actions to use.

**Comment by StelFoog** [comment]
Shouldn't the token that every action has be sufficient, or is that limited only to the active branch?

**Comment by Cactooz** [comment]
> Shouldn't the token that every action has be sufficient, or is that limited only to the active branch?

It might. I had to generate a token myself but if we could grab that token then that is easier 👍🏻

**Comment by StelFoog** [comment]
#40
